api/clients/relay_client.go

```diff
@@ -0,0 +1,206 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	relaygrpc "github.com/Layr-Labs/eigenda/api/grpc/relay"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/hashicorp/go-multierror"
+	"google.golang.org/grpc"
+)
+
+type RelayClientConfig struct {
+	Sockets           map[corev2.RelayKey]string
+	UseSecureGrpcFlag bool
+}
+
+type ChunkRequestByRange struct {
+	BlobKey corev2.BlobKey
+	Start   uint32
+	End     uint32
+}
+
+type ChunkRequestByIndex struct {
+	BlobKey corev2.BlobKey
+	Indexes []uint32
+}
+
+type RelayClient interface {
+	// GetBlob retrieves a blob from a relay
+	GetBlob(ctx context.Context, relayKey corev2.RelayKey, blobKey corev2.BlobKey) ([]byte, error)
+	// GetChunksByRange retrieves blob chunks from a relay by chunk index range
+	// The returned slice has the same length and ordering as the input slice, and the i-th element is the bundle for the i-th request.
+	// Each bundle is a sequence of frames in raw form (i.e., serialized core.Bundle bytearray).
+	GetChunksByRange(ctx context.Context, relayKey corev2.RelayKey, requests []*ChunkRequestByRange) ([][]byte, error)
```

Comment:
Should this client handle deserialization?

Reply:
It's intentionally left serialized so that caller can use the serialized bytes without serializing it again. (node can directly store the bytearrays without the overhead of serializing them again)